### DISCUSSION TITLE
How to force/guide NER detection to produce at least one guess ?

### DISCUSSION THREAD

**Discussion by hwassner** [explosion/spaCy#9628] [Help: Model Advice] [answered]
I'm using Named Entities Dection (a home trained version). I use it on short text lines (titles) where I know that there is as least one NER to find (but I don't know where).

I works quite ok, but sometimes it does not detect any NER in a title. I suppose that the detection confidence was too low...
Since I know that there is one NER in any title, I'd like to acces a prediction even it's unsure.
Is there a way to show the best NER detection ? (event if the confidence is low)

I'm nearly sure that, on my application, this best guess strategy will improve my process.

Any help will be appreciated.

**Comment by adrianeboyd** [comment, marked as answer]
This isn't possible with the `ner` component. With the `spancat` component, you can set a very low threshold for prediction and then select the top N predictions from the stored scores (see https://spacy.io/api/spancategorizer#assigned-attributes), although it's still not guaranteed that the model will make at least one prediction for any given text. This is just brainstorming, but I think you could have some kind of backoff for the case without any predictions, like selecting the first proper noun in the text.
